pkg/trustedresources/verify.go

```diff
@@ -156,14 +174,19 @@ func verifyResource(ctx context.Context, resource metav1.Object, k8s kubernetes.
 		if !passVerification {
 			if p.Spec.Mode == v1alpha1.ModeWarn {
 				logger := logging.FromContext(ctx)
-				logger.Warnf("%w: resource %s in namespace %s fails verification", ErrResourceVerificationFailed, resource.GetName(), resource.GetNamespace())
+				warn = fmt.Errorf("%w: resource %s in namespace %s fails verification", ErrResourceVerificationFailed, resource.GetName(), resource.GetNamespace())
+				logger.Warn(warn)
```

Comment:
ah - I see why you're doing this now. I'd just simplify this above as:

```go
if err != nil {
  type := reasonError
  if p.Spec.Mode == v1alpha1.ModeWarn {
    type = reasonWarn
  }
  return &VerificationError{type: type, err: err}
}

return nil
```

You could probably do this inline with the `verifyInterface` call instead of calling a break.

Reply:
oh I see, thanks! This is much better